### PR TITLE
Image Gallery: Update toolbar position and tooltips

### LINKED ISSUE #35182: [Enhancement]: Add space between a selected image in the gallery and the toolbar
### Describe the solution you'd like

I noticed there's no padding between the image and the toolbar. We need to add it for better clarity and visual order.

image

**Acceptance criteria**

- [ ] There's an 8 px space between the image and the toolbar

### Describe alternatives you've considered

_No response_

### Additional context

**Designs in Figma**

5sAIeTRd9Yp7nSCT33BAWz-fi-5180%3A118552

### LINKED ISSUE #35184: [Enhancement]: Add tooltips for all icons in the image toolbar
### Describe the solution you'd like

For better clarity and accessibility, let's add tooltips when the user hovers over the non-text buttons in the image toolbar.

### Mockup

image

### Acceptance criteria

- [ ] The tooltips are positioned correctly, below the corresponding toolbar button (see https://wordpress.github.io/gutenberg/?path=/story/components-toolbar--toolbars)
- [ ] The drag handle tooltip says `Drag to reorder`
- [ ] The left arrow icon tooltip says `Move left`
- [ ] The right arrow icon tooltip says `Move right`
- [ ] The cover icon tooltip says `Set as cover`
- [ ] The trash icon tooltip says `Remove`

### PR DESCRIPTION
**Pull request by mattsherman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

The position of the toolbar has been adjusted to provide an `8px` margin between the toolbar and the image. The z-index of the toolbar has also been updated to prevent the tooltips from appears behind images.

The toolbar tooltips have also been updated:
- The drag handle tooltip says `Drag to reorder`
- The left arrow icon tooltip says `Move previous`
- The right arrow icon tooltip says `Move next`
- The cover icon tooltip says `Set as cover`
- The trash icon tooltip says `Remove`

Closes #35182
Closes #35184

#### Before

Screenshot 2022-11-09 at 22 37 42

- Note the lack of margin between the image and the toolbar
- Note the lack of the "Remove" tooltip... it is hiding behind the next image

#### After

Screenshot 2022-11-09 at 22 16 50

- Note the margin between the image and the toolbar
- Note the "Remove" tooltip appears properly

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Go to `Products` > `Add new (MVP)`
2. Add a few images to the image gallery
3. Verify that the toolbar appears properly, along with the tooltips

Note: If the image gallery is tested in Storybook, it will not show the tooltips in the correct location, as we are still dependent on an older version of `@wordpress/components` when not running in WordPress (where we use whatever version of `@wordpress/components` it supplies; WordPress 6.1 supplies a version that fixes the tooltip positioning issue).

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/components/src/image-gallery/image-gallery-toolbar.scss [hunk 1]
```diff
@@ -1,7 +1,8 @@
 .woocommerce-image-gallery__toolbar {
 	position: absolute;
-	top: -50px;
+	top: -58px;
 	left: 50%;
 	transform: translateX(-50%);
+	z-index: 1000;
```
Comment (joshuatf) [inline packages/js/components/src/image-gallery/image-gallery-toolbar.scss:6]: Is this an arbitrary number or relative to some other component?  Just curious if it's relative to some other component; if so we could consider using `z-index('.other-component')` but not a blocker for this PR.

**Review by joshuatf — APPROVED** [review]
Testing well and looks much better!  Thanks for this!

Not directly related to this PR and we can handle in a follow-up, but noticed a couple of oddities around the toolbar.  In storybook, it's difficult to see the toolbar in the top row:

Screen Shot 2022-11-14 at 9 59 18 AM


The tooltips appear below the image, far from the toolbar itself:

Screen Shot 2022-11-14 at 10 00 31 AM

File: packages/js/components/src/image-gallery/image-gallery-toolbar.scss (hunk 1 above)
Comment (mattsherman) [reply packages/js/components/src/image-gallery/image-gallery-toolbar.scss:6]: Arbitrary. I tried using the approach you mentioned, but it complained that the other component (the image gallery container element) didn't have a `z-index`. I made a note to circle back to this.

**Comment by mattsherman** [comment]
> Testing well and looks much better! Thanks for this!
> 
> Not directly related to this PR and we can handle in a follow-up, but noticed a couple of oddities around the toolbar. In storybook, it's difficult to see the toolbar in the top row:
> 
> Screen Shot 2022-11-14 at 9 59 18 AM

Yeah, that should ideally be improved. I've made a note.

> The tooltips appear below the image, far from the toolbar itself:
> 
> Screen Shot 2022-11-14 at 10 00 31 AM

Yes, that was mentioned in the PR description...

> Note: If the image gallery is tested in Storybook, it will not show the tooltips in the correct location, as we are still dependent on an older version of @wordpress/components when not running in WordPress (where we use whatever version of @wordpress/components it supplies; WordPress 6.1 supplies a version that fixes the tooltip positioning issue).

I'm not sure how to best handle this issue... because if running with an older version of WordPress (and thus `@wordpress/components`) we will have this bug in the product screen as well.

**Pull request merged**